pydantic/main.py

```diff
@@ -592,6 +603,51 @@ def __repr_args__(self) -> _repr.ReprArgs:
             if not k.startswith('_') and (k not in self.model_fields or self.model_fields[k].repr)
         ]
 
+    def __class_getitem__(cls, item: typing.Union[type[Any], typing.Tuple[type[Any], ...]]) -> type[Any]:
+        def _cache_key(_params: Any) -> typing.Tuple[type[Any], Any, typing.Tuple[Any, ...]]:
+            return cls, _params, typing_extensions.get_args(_params)
+
+        cached = _generic_types_cache.get(_cache_key(item))
+        if cached is not None:
+            return cached
+
+        if not isinstance(item, tuple):
+            item = (item,)
```

Comment:
is it worth having a more descriptive name, e.g.
```suggestion
            typevar_values = (item,)
        else:
            typevar_values = item
```
???

Reply:
yeah, I'm in favor. I noticed that the `__class_getitem__` method is defined on `GenericModel` using `params` as the name of the argument; I renamed it to `item` here following the documentation of `__class_getitem__` but just realized that I think we used `params` as the name in `GenericModel` because that is what it is called in `typing.Generic`. I checked and confirmed that the function `__class_getitem__` is called using an arg, rather than a kwarg, so I was thinking we could maybe just call the argument to the `__class_getitem__` function `typevar_values`.

I think that would be my first preference, since you can then see the intent as soon as you see the function signature, but I could just use your suggestion here if you prefer. (I do think either way is an improvement.)

Reply:
whichever you prefer, it's not a big issue.